### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by andronicus-kim**
## Summary

This PR updates project dependencies

**Related issue**: None

## Changes

Please refer to the nice summary from Copilot below

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: .github/workflows/code-analysis.yml [hunk 1]
```diff
@@ -44,21 +44,3 @@ jobs:
...
-        uses: actions/cache@v4
-        with:
-          path: ~/.gradle/caches/
-          key: gradle-caches-${{ runner.os }}-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
-          restore-keys: |
-            gradle-caches-${{ runner.os }}-
-      - name: ktlint
-        run: ./gradlew ktlintAll
-      - name: Stop Gradle
-        run: ./gradlew --stop
```
Comment (KCeh) [inline .github/workflows/code-analysis.yml:64]: 👍 
I will remove ktlint from expected checks

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -43,6 +41,8 @@ tools-lintapi = { module = "com.android.tools.lint:lint-api", version.ref = "lin
 tools-lintchecks = { module = "com.android.tools.lint:lint-tests", version.ref = "lint" }
 tools-linttests = { module = "com.android.tools.lint:lint-checks", version.ref = "lint" }
 
+ksp-gradle = { module = "com.google.devtools.ksp:symbol-processing-gradle-plugin", version.ref = "ksp" }
```
Comment (KCeh) [inline gradle/libs.versions.toml:46]: Why are we introducing KSP?

File: lint/build.gradle [hunk 3]
```diff
@@ -42,6 +42,7 @@ dependencies {
     testImplementation libs.junit
     testImplementation "org.junit.jupiter:junit-jupiter-api:5.9.2"
     testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:5.9.2"
+    testRuntimeOnly "org.junit.platform:junit-platform-launcher:1.9.2"
```
Comment (KCeh) [inline lint/build.gradle]: Just a hint: we could extract those dependecies to version catalog as well

File: gradle/libs.versions.toml (hunk 2 above)
Comment (KCeh) [reply gradle/libs.versions.toml:46]: OK I see we extracted version 👍

**Review by KCeh — APPROVED** [review]
Looks good 🚀

File: lint/build.gradle (hunk 3 above)
Comment (andronicus-kim) [reply lint/build.gradle]: Absolutely 🤝
